src/ServiceControl/Operations/FailedMessagePersister.cs

```diff
@@ -72,7 +74,7 @@ public FailedMessagePersister(IDocumentStore store, BodyStorageFeature.BodyStora
                 // TODO: Do we need to persist all of these things separately still?
                 message.Headers.MessageIntent(),
                 true,
-                message.Headers.CorrelationId(),
+                message.Headers[Headers.CorrelationId],
```

Comment:
We can't assume that any particular headers exist. Otherwise a missing header throws an exception and the message cannot be ingested.

Reply:
I was trying to find the code in question but couldn't. I will go back to the previous branch Nd see how it was implemented. I planned to point this out on the PR but forgot. Thanks for catching it